src/solutions/ode_solutions.jl

```diff
@@ -78,6 +78,40 @@ function interpolation_residual(sol::AbstractODESolution, t)
   sol(t,Val{1}) - sol.prob.f(sol(t), sol.prob.p, t)
 end
 
+function max_residual_estimate(sol::AbstractODESolution)
```

Comment:
it should probably check `sol.dense` first?

Reply:
Just read the [docs](https://diffeq.sciml.ai/stable/tutorials/ode_example/#Step-3:-Analyzing-the-Solution) for the dense output. It appears that if `dense=false` then we have linear interpolation, which should give a very poor backward error, moreover the derivatives at the nodes will be undefined. So I guess what you're getting at is that we shouldn't check the nodes if `dense=false`?

Reply:
I guess it could just do it, but it won't be the most accurate error estimate in that case.

Reply:
So there are some philosophical issues here. . . IMO the *interpolator is the numerical solution*. Some people think the solution skeleton ((t_i, u_i) pairs) is the solution.

If you adopt my viewpoint, then maybe you'd say we need to do a symmetric derivative at the node? If you adopt the solution skeleton viewpoint, then you might say "only compute the residual between the nodes".

Reply:
Corless gives a definition that might help here:

> Definition 12.1. If the computed solution (typically a piecewise polynomial interpolant) is continuously differentiable, we will call the residual a defect. If the computed solution is only piecewise continuously differentiable, so that the residual is only defined piecewise and may have jump discontinuities, we will call it a deviation.